### PR TITLE
RDoc-1364 OnDirectoryInitialize

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: Documentation/4.1/Raven.Documentation.Pages/server/configuration/storage-configuration.markdown [hunk 1]
```diff
@@ -103,6 +103,40 @@ How many gigabytes of memory should be prefetched before restarting the prefetch
 
 {PANEL/}
 
+{PANEL:Storage.OnCreateDirectory.Exec or Storage.OnDirectoryInitialize.Exec}
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/server/configuration/storage-configuration.markdown]: Please remove `Storage.OnCreateDirectory....`  we just added them in the code to ensure backward compatibility with already released 4.1.3 patch version

File: Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ln -s ~/RavenDB/Server/RavenData/Databases/Northwind/Indexes /mnt/FastDrive/Databases/Northwind/Indexes 
+{CODE-BLOCK/}
+
+{INFO Start RavenDB server _after_ creating soft link to a faster drive mount point: /}
+
+{NOTE/}
+
+{NOTE: Automation}
+
+To help you automate the process, we have added the [Storage.OnDirectoryInitialize](../../server/configuration/storage-configuration#storage.oncreatedirectory.exec-or-storage.ondirectoryinitialize.exec) extension point.
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown]: See above comment and update link

File: Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+Whenever RavenDB creates or opens a directory, it will invoke a process of your choice.
+It allows you to create a script with your own logic, defining juction points as needed.
+
+RavenDB will invoke the process with [optional user arguments](../../server/configuration/storage-configuration#storage.oncreatedirectory.exec.arguments-or-storage.ondirectoryinitialize.exec.arguments) followed by:  
+
+* The environment type (System, Database, Index, Configuration, Compaction)
+* The database name
+* Path of the `DataDir` directory
+* Path of the `Temp` directory
+* Path of the `Journals` directory
```
Comment (arekpalinski) [inline Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown]: Can you provide an example of such script? In particular how to script creation of junction points e.g. 
- Raven runs on D:\ drive and documents data is there but I want all indexes to be on E:\ 
- I want Journals to be on E:\ drive

Comment (arekpalinski) [reply Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown]: Would be great to have the same samples put here in tests so we can validate they work as expected

File: Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+{PANEL: Overview}
+
+{NOTE: Motivation:}
+
+* Avoiding traffic jams.
+
+* Better concurrency.
+
+* Directing each file or directory (e.g. `Voron` files, `Journals`, etc.) to a data storage according to its speed, durability, etc.
```
[the remark this answers is not in the document]
Comment (ayende) [reply Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown]: Voron is the whole storage engine

File: Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown (hunk 3 above)
Comment (iftahbe) [reply Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown]: I don't think we should be the ones supplying these examples. Should be entirely up to the users what they do with this script.

Comment (arekpalinski) [reply Documentation/4.1/Raven.Documentation.Pages/server/storage/customizing-raven-data-files-locations.markdown]: Addressed in https://github.com/ravendb/docs/pull/860/commits/873457593b3168abe7bc348bbf54cdab7e7bfd30

**Pull request merged**